common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -385,13 +386,15 @@ public UTF8String toUpperCase() {
   }
 
   private UTF8String toUpperCaseSlow() {
-    return fromString(toString().toUpperCase());
+    return fromString(toString().toUpperCase(Locale.ROOT));
```

Comment:
I think we shouldn't apply locale when we handle actual data. My take is that the upper lower cases should be locale dependent.

Reply:
Done